### DISCUSSION TITLE
Global app bar / menu bar

### DISCUSSION THREAD

**Discussion by andrejdaskalov** [ghostty-org/ghostty#2642] [Feature Requests, Ideas]
The discussion title is a little ambiguous because I started trying to make one thing and ended up with another.
My initial idea was to try and add global menu support. I know gnome hasn't supported it for a while, but other DEs still do, like KDE Plasma. In fact, I had gnome apps providing menus to my global menu in Plasma already.

However, it seems that only worked up to gtk3, and gtk4 has no such implementation. I was only able to create a menubar, which resides inside the drawn window.
![image](https://github.com/user-attachments/assets/7d0000dc-682c-44be-ad67-706fc2179dad)
> Note: it's a crude demo with only one option

I don't know if anyone would find this useful, but it is natively supported, and could be added as a config option.
If anyone thinks it's useful, let me know.

**Comment by pluiedev** [comment]
I believe this is [PopoverMenuBar](https://docs.gtk.org/gtk4/class.PopoverMenuBar.html). There were talks among us contributors on the GTK side about this but we had conflicting opinions. Perhaps more concrete talking points could win dissenters over? (I personally don't have a strong opinion, but some others do.)

**Comment by 00-kat** [comment]
Forgot to cross-reference it here too, but #3434 is related.
